src/main/java/chess/view/Command.java

```diff
@@ -0,0 +1,79 @@
+package chess.view;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public enum Command {
+
+    START("start"),
+    END("end"),
+    MOVE("move");
+
+    private static final int COMMAND_KEYWORD_INDEX = 0;
+    private static final int START_AND_END_COMMAND_SIZE = 1;
+    private static final int MOVE_COMMAND_SIZE = 3;
+    public static final int MOVE_SOURCE_INDEX = 1;
+    public static final int MOVE_DESTINATION_INDEX = 2;
+    private static final int MOVE_SQUARE_INPUT_SIZE = 2;
+
+    private final String keyword;
+
+    Command(final String keyword) {
+        this.keyword = keyword;
+    }
+
+    public static Command from(final List<String> input) {
+        validateCommand(input);
+        return Arrays.stream(values())
+                .filter(command -> Objects.equals(command.keyword, input.get(COMMAND_KEYWORD_INDEX)))
+                .findFirst()
+                .orElseThrow(CommandException::new);
+    }
+
+    private static void validateCommand(final List<String> command) {
+        if (command.size() == START_AND_END_COMMAND_SIZE) {
+            validateStartAndEnd(command.get(COMMAND_KEYWORD_INDEX));
+            return;
+        }
+        if (command.size() == MOVE_COMMAND_SIZE) {
```

Comment:
```suggestion
        if (isStartOrEndCommand()) {
            validateStartAndEnd(command.get(COMMAND_KEYWORD_INDEX));
            return;
        }
        if (isMoveCommand()) {
```

이런 식으로 추상화 수준을 높여주면 코드 가독성이 더 좋아질 것 같아요.

Reply:
[커맨드 패턴](https://refactoring.guru/ko/design-patterns/command)을 적용해봐도 좋겠네요.

Reply:
시간 안에 새로운 패턴을 적용하지 못할 것 같아 월요일 이후에 적용해보려구요! 말씀해주셔서 감사합니다!